src/roles.js

```diff
@@ -249,29 +250,35 @@ export class Roles {
    * Get the role for device `deviceId`.
    *
    * @param {string} deviceId
+   * @param {object} [options]
+   * @param {number} [options.timeout] TODO docs
    * @returns {Promise<Role>}
    */
-  async getRole(deviceId) {
-    /** @type {string} */
-    let roleId
-    try {
-      const roleAssignment = await this.#dataType.getByDocId(deviceId)
-      roleId = roleAssignment.roleId
-    } catch (e) {
-      // The project creator will have the creator role
-      const authCoreId = await this.#coreOwnership.getCoreId(deviceId, 'auth')
-      if (authCoreId === this.#projectCreatorAuthCoreId) {
-        return CREATOR_ROLE
-      } else {
-        // When no role assignment exists, e.g. a newly added device which has
-        // not yet synced role records.
-        return NO_ROLE
-      }
+  async getRole(deviceId, options = {}) {
```

Comment:
I chose to add new behavior to `Roles`, as I felt it was a clean place to put it. Let me know if that seems wrong.